src/classes/HH_Container_LCTRL.cls

```diff
@@ -271,13 +272,24 @@ public with sharing class HH_Container_LCTRL {
             'MailingCountry', 
             'MailingLatitude', 
             'MailingLongitude'};
-        if (UTIL_Describe.isMiddleNameEnabled()) {
-            setContactFields.add('Middlename');
-        }
+
+        addContactMiddleNameField(setContactFields);
+
         return setContactFields;
     }
 
     /*******************************************************************************************************
+    * @description Adds MiddleName field into the set of field names
+    * @param fieldNames Set of field names 
+    * @return void
+    */
+    private static void addContactMiddleNameField(Set<String> fieldNames) {        
+        if (UTIL_Describe.isMiddleNameEnabled()) {
+            fieldNames.add('MiddleName');
```

Comment:
@RadSFDeveloper The only real change here is the switch from `Middlename` to `MiddleName`. The Set is Case Sensitive and the custom household name value is a open text field, so the User can simply enter the following value and cause this to still fail:

`{!{!Salutation} {!FirstName} {!Middlename}} {!LastName} Household`

I think the right answer here is to verify that "middlename" (in any _case_) is not already in the list of fields. Without looking at the code, I'm not 100% sure where the custom naming fields are added to the set - before or after the getContactFields() call. So either this method needs to do that check, or where ever the logic is to add the specific fields listed in the custom naming template.